dashboard/lib/services/i18n/curriculum_sync_utils/sync_out.rb

```diff
@@ -58,16 +58,22 @@ def self.postprocess(result)
           if result.key?(:lessons)
             rekeyed_lessons = result[:lessons].map do |lesson_url, lesson_data|
               route_params = Rails.application.routes.recognize_path(lesson_url)
-              lesson = Lesson.joins(:script).
-                find_by(
+              # Filter for only lessons which are "numbered".
+              # Only these lessons guarantee that their relative position is unique.
+              lessons = Lesson.joins(:script).
+                where(
                   "scripts.name": route_params[:script_id],
                   relative_position: route_params[:position].to_i,
-                  has_lesson_plan: true
-                )
-              unless lesson.present?
-                STDERR.puts "could not find lesson for url #{lesson_url.inspect}"
+                ).select(&:numbered_lesson?)
+              if lessons.count == 0
+                STDERR.puts "Could not find lesson for url #{lesson_url.inspect}"
                 next
               end
+              if lessons.count > 1
+                STDERR.puts "More than one lesson found for url #{lesson_url.inspect}. This should be investigated."
+                next
+              end
+              lesson = lessons.first
```

Comment:
Is it ever possible the first lesson isn't necessarily the lesson we want? Maybe a newer version also exists or something?

Reply:
Ah missed that sorry